storage/data-plane/src/main/java/com/microsoft/azure/storage/blob/PipelineOptions.java

```diff
@@ -41,25 +41,26 @@
      * For more samples, please see the [Samples file](%https://github.com/Azure/azure-storage-java/blob/master/src/test/java/com/microsoft/azure/storage/Samples.java)
      */
     public PipelineOptions() {
-        this.logger = new HttpPipelineLogger() {
-            @Override
-            public HttpPipelineLogLevel minimumLogLevel() {
-                return HttpPipelineLogLevel.OFF;
-            }
+        this.logger = new HttpPipelineLoggerImpl();
```

Comment:
Why did this change? I don't see this in the spotbugs report.

Reply:
I have seen this in spotbugs report, make the private inner class static one. I run the command under central package.

> At PipelineOptions.java:[line 46]
> Field com.microsoft.azure.storage.blob.PipelineOptions.logger
> The class com.microsoft.azure.storage.blob.PipelineOptions$1 could be refactored into a named _static_ inner class
> This class is an inner class, but does not use its embedded reference to the object which created it.  This reference makes the instances of the class larger, and may keep the reference to the creator object alive longer than necessary.  If possible, the class should be made into a static inner class. Since anonymous inner classes cannot be marked as static, doing this will require refactoring the inner class so that it is a named inner class.